src/components/structures/ThreadPanel.tsx

```diff
@@ -305,7 +305,7 @@ const ThreadPanel: React.FC<IProps> = ({ roomId, onClose, permalinkCreator }) =>
                 <Measured sensor={card.current} onMeasurement={setNarrow} />
                 {timelineSet ? (
                     <TimelinePanel
-                        key={timelineSet.getFilter()?.filterId ?? roomId + ":" + filterOption}
+                        key={filterOption + ":" + (timelineSet.getFilter()?.filterId ?? roomId)}
```

Comment:
Seems like we should add a test for the problem that this fixes. Make sure the filtered view shows a subset of what all threads shows.

Was there a warning about duplicate keys being used? Can we fail tests if this ever occurs?

Reply:
We do have a warning for this, it's the "timeline panel being reused with a different timelineset" warning. I don't think there's an easy way to reliably automate testing this, though.

Reply:
> We could add a test to ensure the filtered view shows a subset of what all threads shows.

What about a test for this scenario ^?

> We do have a warning for this, it's the "timeline panel being reused with a different timelineset" warning.

Where is this warning emitted? Do you mean [`Replacing timelineSet on a TimelinePanel - confusion may ensue`](https://github.com/matrix-org/matrix-react-sdk/blob/aeb9f4373fff7cc8e1fa0e2c26f870dc50119cae/src/components/structures/TimelinePanel.tsx#L336)?

Reply:
Yes, that's exactly the warning I meant. Sorry, I'm on a personal device right now, going by memory.

I'll reply in more detail later when I'm back at work.

Reply:
@justjanne Friendly poke

We can write an end-to-end Cypress test for this particular case which avoids the internal hell implementation and the current test harness for it.

---

If the [`Replacing timelineSet on a TimelinePanel - confusion may ensue`](https://github.com/matrix-org/matrix-react-sdk/blob/aeb9f4373fff7cc8e1fa0e2c26f870dc50119cae/src/components/structures/TimelinePanel.tsx#L336) warning is always considered a bug when we hit that point, we should do more to ask for a bug report when it happens. We can show a bug report suggestion in the thread view and explain that something went wrong.